programs/mpl-core/src/plugins/update_delegate.rs

```diff
@@ -48,6 +48,23 @@ impl DataBlob for UpdateDelegate {
 }
 
 impl PluginValidation for UpdateDelegate {
+    fn validate_create(
+        &self,
+        ctx: &PluginValidationContext,
+    ) -> Result<ValidationResult, ProgramError> {
+        if let Some(resolved_authorities) = ctx.resolved_authorities {
+            if resolved_authorities.contains(ctx.self_authority) {
+                return approve!();
+            }
+        }
+
+        if self.additional_delegates.contains(ctx.authority_info.key) {
```

Comment:
I don't think this covers the case where the payer is the auth. I think that's somewhere in the plugin validation context.

Reply:
`resolve_authority` gets called at the beginning of `process_create`, and whichever `AccountInfo` is chosen there is put into the `PluginValidationContext`.  Isn't that correct?

Reply:
You're right. My brain saw ctx.authority_info and thought it was was the accounts struct. My bad.